code/modules/clothing/glasses/engine_goggles.dm

```diff
@@ -157,8 +160,76 @@
 
 	modes = list(MODE_NONE = MODE_SHUTTLE, MODE_SHUTTLE = MODE_NONE)
 
+
+/obj/item/clothing/glasses/meson/engine/atmos_imaging
+	name = "atmospheric thermal imaging goggles"
+	desc = "Goggles used by Atmospheric Technician to see the thermal energy of gasses in open areas"
+	icon_state = "trayson-radiation"
+	inhand_icon_state = "trayson-meson"
+	glass_colour_type = /datum/client_colour/glass_colour/gray
+
+	modes = list(MODE_NONE = MODE_ATMOS_THERMAL, MODE_ATMOS_THERMAL = MODE_NONE)
+
+/obj/item/clothing/glasses/meson/engine/atmos_imaging/toggle_mode(mob/user, voluntary)
+	mode = modes[mode]
+	to_chat(user, "<span class='[voluntary ? "notice":"warning"]'>[voluntary ? "You turn the goggles":"The goggles turn"] [mode ? "to [mode] mode":"off"][voluntary ? ".":"!"]</span>")
+	switch(mode)
+		if(MODE_ATMOS_THERMAL)
+			change_glass_color(user, /datum/client_colour/glass_colour/lightorange)
+		if(MODE_NONE)
+			change_glass_color(user, initial(glass_colour_type))
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(H.glasses == src)
+			H.update_sight()
+
+	update_appearance()
+	update_action_buttons()
+
+/obj/item/clothing/glasses/meson/engine/atmos_imaging/update_icon_state()
+	icon_state = inhand_icon_state = "trayson-[mode]"
+	return ..()
+
+/obj/item/clothing/glasses/meson/engine/atmos_imaging/process()
+	if(!ishuman(loc))
+		return
+	var/mob/living/carbon/human/user = loc
+	if(user.glasses != src || !user.client)
+		return
+	if(mode == MODE_ATMOS_THERMAL)
+		atmos_thermal(user)
+
+/proc/atmos_thermal(mob/viewer, range = 4, duration = 10)
+	if(!ismob(viewer) || !viewer.client)
+		return
+	for(var/turf/open in view(range, viewer))
+		if(open.blocks_air)
+			continue
+		var/calc
+		var/datum/gas_mixture/environment = open.return_air()
+		var/temp = round(environment.temperature)
+		var/image/pic = image('icons/turf/overlays.dmi', open, "greyOverlay", ABOVE_ALL_MOB_LAYER)
+		if(temp <= 273.15)
+			pic.color = COLOR_STRONG_BLUE
+		if(temp > 273.15 && temp <= 283.15)
+			calc = max(round(temp/280, 0.01), 0)
+			pic.color = BlendRGB(COLOR_DARK_CYAN, COLOR_LIME, calc)
+		else if(temp >= 100)
+			pic.color = COLOR_RED
+		else if(temp > 280 && temp <= 1000)
+			calc =    clamp(round((temp-280)/720, 0.01), 0, 1)
+			pic.color = BlendRGB(COLOR_LIME, COLOR_YELLOW, calc)
+		else
+			calc =    clamp(round((temp-1000)/4000, 0.01), 0, 1)
+			pic.color = BlendRGB(COLOR_YELLOW, COLOR_RED, calc)
```

Comment:
maybe this would be better as a ``switch()``?

Reply:
They didn't write this section of the code, I did, and this isn't the most up-to-date version.
The most up-to-date one uses defines for all the numbers specifically to make it easier for them to change to what they wanted without breaking like they have it now.
You're right though, a switch case would probably be best. I can't say I can edit their code from my computer though.
If you want to see the newest one just ping me in discord I'll grab it for you